### PR TITLE
feat(frame): better log for proxy_protocol config disabled

### PR DESCRIPTION
**Pull request by JimMoen**
https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt

Packet header `PROXY TCP...`

### CODE REVIEW

File: src/emqx_frame.erl [hunk 1]
```diff
@@ -100,6 +100,8 @@ parse(Bin) ->
 -spec(parse(binary(), parse_state()) -> parse_result()).
 parse(<<>>, {none, Options}) ->
     {more, {none, Options}};
+parse(<<"PROXY ", _Rest/binary>>, _State) ->
+    error(proxy_protocol_setting_invalid);
```
Comment (tigercl) [inline src/emqx_frame.erl:104]: Add log for this error

**Comment by HJianBo** [comment]
Do we need to probe the proxy-protocol v2 in this pr?

**Comment by JimMoen** [comment]
> Do we need to probe the proxy-protocol v2 in this pr?

added.

**Review by terry-xiaoyu — APPROVED** [review]

File: src/emqx_frame.erl [hunk 2]
```diff
@@ -100,6 +111,13 @@ parse(Bin) ->
 -spec(parse(binary(), parse_state()) -> parse_result()).
 parse(<<>>, {none, Options}) ->
     {more, {none, Options}};
+parse(<<?PROXY_TCP, IPVer:8, " ", _Rest/binary>>, _State)
```
Comment (zmstone) [inline src/emqx_frame.erl:114]: match `none` state 
Also for the other three classes

File: src/emqx_frame.erl [hunk 3]
```diff
@@ -100,6 +111,13 @@ parse(Bin) ->
...
+  when IPVer =:= 4 orelse IPVer =:= 6 ->
```
Comment (zmstone) [inline src/emqx_frame.erl:115]:
```suggestion
  when IPVer =:= $4 orelse IPVer =:= $6 ->
```

File: src/emqx_connection.erl [hunk 4]
```diff
@@ -629,10 +629,15 @@ parse_incoming(Data, Packets, State = #state{parse_state = ParseState}) ->
             NState = State#state{parse_state = NParseState},
             parse_incoming(Rest, [Packet|Packets], NState)
     catch
+        error:proxy_protocol_config_disabled:Stk ->
```
Comment (zmstone) [inline src/emqx_connection.erl:632]: stacktrace is not important here.

File: src/emqx_connection.erl [hunk 5]
```diff
@@ -629,10 +629,14 @@ parse_incoming(Data, Packets, State = #state{parse_state = ParseState}) ->
             NState = State#state{parse_state = NParseState},
             parse_incoming(Rest, [Packet|Packets], NState)
     catch
+        error:proxy_protocol_config_disabled:_Stk ->
+            ?LOG(error,
+                 "~nParse failed for proxy_protocol packet~nFrame data:~0p~n", [Data]),
```
Comment (zmstone) [inline src/emqx_connection.erl:634]:
```suggestion
                 "~nMalfformed packets, please set "listener.NAME.ZONE.proxy_protocol" to true",
```
even better if `NAME` and `ZONE` is known here, then we can print out the exact config key like: `listener.tcp.external.proxy_protocol`

**Review by zmstone — CHANGES_REQUESTED** [review]

File: src/emqx_connection.erl (hunk 5 above)
Comment (JimMoen) [reply src/emqx_connection.erl:634]: Just LOG.
No `Stk` and `Packet Data`

Comment (zmstone) [reply src/emqx_connection.erl:634]: Yes. since we can be 100% sure it's a misconfiguration, the logs should tell the user exactly what's wrong.

Comment (JimMoen) [reply src/emqx_connection.erl:634]: Seems doesn't have a good way to get `NAME` and `ZONE` of the current packet.
So just log as
`~nMalfformed packet, please check proxy_protocol config for current listener and zone~n`

**Review by zmstone — APPROVED** [review]

**Review by tigercl — APPROVED** [review]

**Review by terry-xiaoyu — APPROVED** [review]

**Pull request merged**